fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -1092,43 +1106,108 @@ private CompletableFuture<Integer> moveDocsFromPartition(@Nonnull final LuceneRe
+                }
+                if (timings.createPartitionNanos > 0) {
+                    logMessage.addKeyAndValue("createPartitionMicros", TimeUnit.NANOSECONDS.toMicros(timings.createPartitionNanos - endCleanupNanos));
+                }
+                logMessage.addKeyAndValue("updateMicros", TimeUnit.NANOSECONDS.toMicros(updateNanos - updateStart));
+                if (timerSnapshot != null && state.context.getTimer() != null) {
+                    logMessage.addKeysAndValues(
+                            StoreTimer.getDifference(state.context.getTimer(), timerSnapshot)
+                                    .getKeysAndValues());
+                }
+                LOGGER.debug(logMessage.toString());
+            }
+            return records.size();
         });
     }
 
+    @Nonnull
+    private CompletableFuture<Integer> removeEmptyPartition(@Nonnull final LuceneRepartitionPlanner.RepartitioningContext repartitioningContext) {
+        // sanity check
+        if (repartitioningContext.countToMove != 0) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("removeEmptyPartition called with invalid countToMove {}", repartitioningContext.countToMove);
+            }
+            return CompletableFuture.completedFuture(0);
+        }
+        if ((repartitioningContext.olderPartition == null) && (repartitioningContext.newerPartition == null)) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("removeEmptyPartition called with null neighboring partitions");
+            }
+            return CompletableFuture.completedFuture(0);
+        }
+        RepartitionTimings timings = new RepartitionTimings();
+        final StoreTimerSnapshot timerSnapshot = getStoreTimerSnapshot();
+        timings.startNanos = System.nanoTime();
+
+        final LucenePartitionInfoProto.LucenePartitionInfo partitionInfo = repartitioningContext.sourcePartition;
+        final Tuple groupingKey = repartitioningContext.groupingKey;
+
+        // reset partition info for deleted partition
+        state.context.ensureActive().clear(partitionMetadataKeyFromPartitioningValue(groupingKey, getPartitionKey(partitionInfo)));
+        timings.clearInfoNanos = System.nanoTime();
+
+        // Clear empty data area
+        Range partitionDataRange = Range.startsWith(state.indexSubspace.subspace(groupingKey.add(PARTITION_DATA_SUBSPACE).add(partitionInfo.getId())).pack());
+        state.context.clear(partitionDataRange);
+        timings.emptyingNanos = System.nanoTime();
+
+        if (repartitioningContext.olderPartition != null) {
+            // update other partition's metadata (set "to" from deleted partition)
+            LucenePartitionInfoProto.LucenePartitionInfo.Builder builder = repartitioningContext.olderPartition.toBuilder();
+            builder.setTo(ByteString.copyFrom(partitionInfo.getTo().toByteArray())); // ensure we get a copy of the original bytes
```

Comment:
I'm not sure why you need to copy.
```suggestion
            builder.setTo(partitionInfo.getTo()));
```